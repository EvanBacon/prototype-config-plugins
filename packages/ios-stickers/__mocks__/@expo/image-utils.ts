export const generateImageAsync = jest.fn(async (input: any, { src }) => {
  const fs = require("fs");
  return { source: fs.readFileSync(src) };
});

export const compositeImagesAsync = jest.fn(async ({ foreground }) => {
  return foreground;
});
