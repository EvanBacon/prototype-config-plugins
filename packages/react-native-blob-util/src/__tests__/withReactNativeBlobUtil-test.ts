import { XML } from "@expo/config-plugins";

import { appendDownloadCompleteAction } from "../withReactNativeBlobUtil";

const fixture = `
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.bacon.yolo68">
  <application android:name=".MainApplication" android:label="@string/app_name" android:icon="@mipmap/ic_launcher" android:roundIcon="@mipmap/ic_launcher_round" android:allowBackup="true" android:theme="@style/AppTheme" android:usesCleartextTraffic="true">
    <activity android:name=".MainActivity" android:label="@string/app_name" android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode" android:launchMode="singleTask" android:windowSoftInputMode="adjustResize" android:theme="@style/Theme.App.SplashScreen" android:screenOrientation="portrait">
      <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <category android:name="android.intent.category.LAUNCHER"/>
      </intent-filter>
      <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <data android:scheme="com.bacon.yolo68"/>
      </intent-filter>
    </activity>
    <activity android:name="com.facebook.react.devsupport.DevSettingsActivity"/>
  </application>
</manifest>`;

describe(appendDownloadCompleteAction, () => {
  it("should return manifest", async () => {
    const manifest = (await XML.parseXMLAsync(fixture)) as any;
    appendDownloadCompleteAction(manifest);
    const firstResults = XML.format(manifest);
    expect(firstResults).toMatchSnapshot();
    expect(firstResults).toMatch(
      /<action android:name="android\.intent\.action\.DOWNLOAD_COMPLETE"\/>/
    );

    // Doesn't mutate original.
    appendDownloadCompleteAction(manifest);
    const secondResults = XML.format(manifest);
    expect(secondResults).toMatch(firstResults);
  });
});
