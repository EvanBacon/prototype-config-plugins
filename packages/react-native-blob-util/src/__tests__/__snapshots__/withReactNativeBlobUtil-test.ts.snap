// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appendDownloadCompleteAction should return manifest 1`] = `
"<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.bacon.yolo68">
  <application android:name=".MainApplication" android:label="@string/app_name" android:icon="@mipmap/ic_launcher" android:roundIcon="@mipmap/ic_launcher_round" android:allowBackup="true" android:theme="@style/AppTheme" android:usesCleartextTraffic="true">
    <activity android:name=".MainActivity" android:label="@string/app_name" android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode" android:launchMode="singleTask" android:windowSoftInputMode="adjustResize" android:theme="@style/Theme.App.SplashScreen" android:screenOrientation="portrait">
      <intent-filter>
        <action android:name="android.intent.action.MAIN"/>
        <action android:name="android.intent.action.DOWNLOAD_COMPLETE"/>
        <category android:name="android.intent.category.LAUNCHER"/>
      </intent-filter>
      <intent-filter>
        <action android:name="android.intent.action.VIEW"/>
        <category android:name="android.intent.category.DEFAULT"/>
        <category android:name="android.intent.category.BROWSABLE"/>
        <data android:scheme="com.bacon.yolo68"/>
      </intent-filter>
    </activity>
    <activity android:name="com.facebook.react.devsupport.DevSettingsActivity"/>
  </application>
</manifest>"
`;
